### PR TITLE
LPD-25265 - save search keywords even when submitting for role permissions changes

### PR DESCRIPTION
**Pull request by wanderlast**
Hi @smottal,

https://liferay.atlassian.net/browse/LPD-25265, tied to LPP.

This is a fix for LPD-25265 where search results are not kept when changes are submitted. This change does mean that **search results will be kept until manually cleared using the clear button**. The change saves the keywords as its own variable in the PortletConfigurationDisplayContext and adds it to the updateRolePermissionsURL. I checked and I don't believe this URL is used anywhere else, so this theoretically shouldn't change the behavior anywhere else, but this also does mean that the behavior is unique to here.

### CODE REVIEW

**Comment by wanderlast** [comment]
ci:test:sf

File: modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/display/context/PortletConfigurationPermissionsDisplayContext.java [hunk 1]
```diff
@@ -367,6 +367,8 @@ public SearchContainer<Role> getRoleSearchContainer() throws Exception {
 		RoleSearchTerms searchTerms =
 			(RoleSearchTerms)roleSearchContainer.getSearchTerms();
 
+		setKeywords(searchTerms.getKeywords());
```
[the remark this answers is not in the document]
Comment (smottal) [reply modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/display/context/PortletConfigurationPermissionsDisplayContext.java:370]: or you can use the set to fill with this value OR, if null, the one from the request so you don't need the get method but you can use directly _keywords

I don't know if it works but you should try because I don't like that here we are settings something but from the get we can receive a different value

**Comment by smottal** [comment]
@wanderlast can you also create a functional test for this? Thx

**Comment by wanderlast** [comment]
Hi @smottal,

I've made the requested changes and added the functional test as requested as well.

**Comment by brianikim** [comment]
ci:test:playwright

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
Hello @wanderlast , please make sure you fix the SF error.

File: modules/test/playwright/fixtures/portletConfigurationPermissionsPagesTest.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
```
Comment (smottal) [inline modules/test/playwright/fixtures/portletConfigurationPermissionsPagesTest.ts:2]: 2024 I think

File: modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
```
Comment (smottal) [inline modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts:2]: 2024

File: modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        this.permissionsMenuItem = page.getByRole('menuitem', { name: 'Permissions' });
+        this.productMenuPage = new ProductMenuPage(page);
+
+        this.clearLink = this.permissionsFrame.getByLabel('Clear');
+        this.ownerRoleCell = this.permissionsFrame.getByRole('cell', {name : 'Owner'});
+        this.resultsBanner = this.permissionsFrame.getByText('Results Found for');
+        this.saveButton = this.permissionsFrame.getByRole('button', { name: 'Save' });
+        this.searchBar = this.permissionsFrame.getByPlaceholder('Search for');
+        this.siteMemberRoleCell = this.permissionsFrame.getByRole('cell', {name : 'Site Member'});
+        this.successMessage = this.permissionsFrame.getByText('Success:Your request completed successfully.');
```
Comment (smottal) [inline modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts:39]: I think this can be replaced by an utility already available to check successMessage (search for `waitForSuccessAlert`)

File: modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    async goto() {
+        await this.productMenuPage.goToPages();
+        await this.pageOptionsMenu.first().click();
+        await this.permissionsMenuItem.click();
+    }
+
+    async switchDeltaFrom20To4() {
```
Comment (smottal) [inline modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts:48]: can we rename this to changePagination and have it dynamic? Like having the current number (20) and the desired one (4) as parameters

File: modules/test/playwright/tests/portlet-configuration-web/config.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/config.ts:2]: 2024

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+test ('LPD-25265 search results should stay when form submitted', async ({
+    page,
+    portletConfigurationPermissionsPage,
+}) => {
+    await portletConfigurationPermissionsPage.goto();
+    await expect(
+        portletConfigurationPermissionsPage.searchBar
+    ).toBeVisible();
+    await page.waitForTimeout(500);
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:26]: why is this needed?

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    
+    await portletConfigurationPermissionsPage.searchBar.click();
+    await portletConfigurationPermissionsPage.searchBar.fill('r');
+    await portletConfigurationPermissionsPage.searchBar.press('Enter');
+    await expect(
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:31]: new empty line before this

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        portletConfigurationPermissionsPage.resultsBanner
+    ).toBeVisible();
+    await expect(portletConfigurationPermissionsPage.ownerRoleCell).toBeVisible();
+    await expect(portletConfigurationPermissionsPage.siteMemberRoleCell).toBeVisible();
+
+    await portletConfigurationPermissionsPage.switchDeltaFrom20To4();
+    await expect(portletConfigurationPermissionsPage.ownerRoleCell).toBeVisible();
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:38]: new empty line before this

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    await expect(portletConfigurationPermissionsPage.siteMemberRoleCell).toHaveCount(0);
+    
+    await portletConfigurationPermissionsPage.saveButton.click();
+	await expect(portletConfigurationPermissionsPage.successMessage).toBeVisible();
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:42]: indent error and new empty line before this

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:43]: no empty line (all the expect in the same "block")

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    await expect(portletConfigurationPermissionsPage.resultsBanner).toBeVisible();
+    await expect(portletConfigurationPermissionsPage.ownerRoleCell).toBeVisible();
+    await expect(portletConfigurationPermissionsPage.siteMemberRoleCell).toHaveCount(0);
+
+    await portletConfigurationPermissionsPage.clearLink.click();
+    await expect(portletConfigurationPermissionsPage.siteMemberRoleCell).toBeVisible();
```
Comment (smottal) [inline modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:49]: new empty line before this

**Review by smottal — CHANGES_REQUESTED** [review]
Beacuse it's a new playwright module I thin you need to add also a test.properties inside the folder file to indicate the component and to update the test.properties file in the portal root to update `playwright.test.project[playwright-js-tomcat90-mysql57-jdk8][um]=\` to include also this new project (check in this PR https://github.com/brianchandotcom/liferay-portal/pull/149977/files#diff-0dc5a96f7041113b3e2e0777ee0f3118e2b6bb53b4bd82a73ebd5228e8ee4da7)

**Comment by wanderlast** [comment]
ci:test:sf

File: modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts (hunk 4 above)
Comment (wanderlast) [reply modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts:39]: I actually tried to use this previously but wasn't able to pass in the frame successfully (so it would never find the success alert). I'll try again, but I'm not sure it would work in this case.

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts (hunk 7 above)
Comment (wanderlast) [reply modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:26]: I was experiencing some issues with the modal loading occasionally taking a really long time and the test failing due to it. I'll remove if you prefer.

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
Hi @smottal,

Thanks for looking at this. I made all requested changes except the one noted earlier (which is the change to calling waitForSuccessAlert). The modal doesn't close when we save, so I don't believe I can properly send in a page variable to the function and still have it pass. Every time I've tried to run while using it, it has failed and not found the success alert despite it clearly showing up on the screenshot timeline.

File: modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts (hunk 11 above)
Comment (smottal) [reply modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:44]: I think this one is still missing

Comment (wanderlast) [reply modules/test/playwright/tests/portlet-configuration-web/rolePermissions.spec.ts:42]: Sorry about that, you're right, please see: dd1a79e96a3c9b547de7829c48653789d0906bca

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
ci:test:sf

**Comment by wanderlast** [comment]
ci:test:sf

File: modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+		);
+	}
+
+	async goto() {
+		await this.productMenuPage.goToPages();
+		await this.pageOptionsMenu.first().click();
+		await this.permissionsMenuItem.click();
+	}
+
+	async changePagination(startValue: number, endValue: number) {
```
Comment (smottal) [inline modules/test/playwright/pages/portlet-configuration-web/PortletConfigurationPermissionsPage.ts:61]: 👍

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:um

**Comment by smottal** [comment]
@wanderlast can you check the test because in the UM suite is failing

**Comment by wanderlast** [comment]
Unless I'm missing something, all of the failures seem to be due to some kind of network failure:

`     [echo] Target exception: com.liferay.jenkins.results.parser.GitWorkingDirectory$GitWorkingDirectoryRuntimeException: Runtime exception  occurred in liferay-portal
     [echo] Unable to checkout master-temp-1715884992699
     [echo] error: unable to read sha1 file of portal-web/test/functional/com/liferay/portalweb/dependencies/test-app-portlet-7.2.0.war/WEB-INF/classes/com/liferay/testhook/hook/model/impl/TestHookUserImpl.class (2d07ed9b6eb695e518a5c310a2ad2ac6f5841985)`

`Caused by: java.net.UnknownHostException: mirrors-no-cache.lax.liferay.com`

`Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: no cgroup mount found in mountinfo: unknown.`

**Comment by wanderlast** [comment]
ci:test:um

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward
